openapi/sms/v1/common.yaml

```diff
@@ -0,0 +1,120 @@
+    Error:
+      type: object
+      properties:
+        message:
+          type: string
+          example: 'Unable to connect to the Partner service.'
+          description: 'Description of the error'
+      required:
+        - message
+    InvalidFormatError:
+      allOf:
+        - $ref: '#/components/schemas/Error'
+        - type: object
+          properties:
+            keyword:
+              type: string
+              example: 'Required'
+            dataPath:
+              type: string
+              example: '.someArray[0]'
+            params:
+              type: object
+              example: { 'missingProperty': 'messageId' }
+            message:
+              type: string
+              example: "should have required property 'messageId'"
+    InvalidFormatErrorBatch:
+      type: object
+      properties:
+        errors:
+          type: array
+          items:
+            $ref: '#/components/schemas/InvalidFormatError'
+            minimum: 1
+            maximum: 1000
+    Message:
+      type: object
+      properties:
+        source:
+          type: string
+          pattern: '^[A-Za-z0-9]+$'
+          example: '436771234567'
+          description: 'Source (Sender ID) of the message'
+        destination:
+          type: string
+          pattern: '^\d+$'
+          example: '381641234567'
+          description: 'MSISDN of the receiver of the message'
+        message:
+          type: string
+          example: 'Hello world!'
+          description: 'Message content (body)'
+      required:
+        - source
+        - destination
+        - message
+    IdentifiableMessage:
+      allOf:
+        - $ref: '#/components/schemas/Message'
+        - type: object
+          properties:
+            messageId:
+              type: string
+              example: 'WI0ksjocu3b8W8kQU26M'
+              description: 'Unique message identifier'
+          required:
+            - messageId
+  parameters:
+    clientIdPathParam:
+      name: clientId
+      in: path
+      required: true
+      description: Ths is the ID used to reference the client user's integration with the Partner service. It is generated when the client user completes the onboarding.
+      example: 39a281c7-942f-49a9-91c2-8f0dba750f7f
+      schema:
+        type: string
+        format: uuid
+  responses:
+    AcceptedResponse:
+      description: Request was accepted
+    InvalidFormatResponse:
+      description: Request body is invalid and does not follow API specification
+      content:
+        application/json:
+          schema:
+            $ref: './common.yaml#/components/schemas/InvalidFormatErrorBatch'
+    AuthenticationFailedResponse:
+      description: Authentication failed - Invalid or missing OAuth token
+    AuthorizationFailedResponse:
+      description: Authorization failed - OAuth client ID does not belong to a registered partner
+    ClientDoesNotExistResponse:
+      description: Client does not exists
```

Comment:
Previously, we had the following description in case of 404:
"_Client does not have a configuration recorded within the Partner service. All messages will be marked as failed._"
It is good that you removed the second sentence because we think that this is an integration error, and as we understand the current behaviour with @pendicg24 , the messages should remain in the queue in that case until the error is fixed and then they are resent.
Do we understand it correctly?

Reply:
This is a fair conclusion to make, though my understanding was that any action that resulted with a non-existent client meant that we do nothing (not queue anything at all) and simply return 404. I'm not sure what the best approach here is 🤔